src/LibraryManager/Configuration/ProxySettings.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using Microsoft.Web.LibraryManager.Contracts.Configuration;
+
+namespace Microsoft.Web.LibraryManager.Configuration
+{
+    /// <summary>
+    /// Determine the proxy settings to use for an HTTP request
+    /// </summary>
+    public class ProxySettings
+    {
+        private readonly ISettings _settings;
+
+        /// <summary>
+        /// Returns a ProxySettings that uses the default configuration file.
+        /// </summary>
+        public static ProxySettings Default => new ProxySettings(new Settings());
+
+        /// <summary>
+        /// Create a ProxySettings using the specified ISettings instance
+        /// </summary>
+        /// <param name="settings"></param>
+        public ProxySettings(ISettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// Get the proxy to use for a given URI
+        /// </summary>
+        public IWebProxy GetProxy(Uri uri)
+        {
+            bool bypassProxy = CheckForProxyBypass(uri);
+
+            return bypassProxy ? null : GetUserConfiguredProxy();
+        }
+
+        private bool CheckForProxyBypass(Uri uri)
+        {
+            if (_settings.TryGetValue(Constants.HttpProxyBypass, out string bypassValue))
+            {
+                string[] bypasses = bypassValue.Split(';');
+                foreach (string bypass in bypasses)
+                {
+                    if (new Uri(bypass).IsBaseOf(uri))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private IWebProxy GetUserConfiguredProxy()
```

Comment:
If these settings were to be exposed e.g. in the VS settings UI, then we would also need that UI to have a cache invalidation strategy.

Since the HttpClient is already being cached for the host, this method should only be called once per host.  See my other comment about the technicalities of determining a per-host proxy.

Reply:
Sure, sounds fine.